### DISCUSSION TITLE
Repo switch on to changed Username of same Owner

### DISCUSSION THREAD

**Discussion by RVanVlietII** [community/community#54624] [Repositories] [answered]
### Select Topic Area

Show & Tell

### Body

When you change your username, how do you switch your repo's location on the terminal (for mac users) so that the repo can be saved with push and pull origin commands? The repo is still reading on the old username but needs to be switched to the new username even though its still the same owner.

**Comment by airtower-luna** [comment]
You can update the *remote URL* with [`git remote set-url`](https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-urlem). Usually the remote will be named `origin` (it's the default name when you clone a repository), if you're not sure `git remote -v` will list all your remotes and their current URLs.

The command will look similar to:

```
git remote set-url origin NEW_URL
```

Mind that you'll need to do this for each affected repository.

**Reply by RVanVlietII** [reply]
So do I copy the new url from my github profile and paste it as the NEW_URL when doing the command to change the repo's save location?

**Reply by airtower-luna** [reply]
Yes, exactly. That's assuming your remote for GitHub is named `origin`, if it has another name you'll need to adjust that too.

If you're not familiar with what a "remote" is, I recommend you read [Git Basics - Working with Remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes).

**Reply by RVanVlietII** [reply]
So I understand what that means. It shows my fetch and push URL's as the correct new one containing the new RVanVlietII username. But when I to push or pull data to the origin main on the terminal, it requests my GitHub username and password and every time I put the password for the RVanVlietII username, it states passwords aren't used anymore and I must use a token. And then the file doesn't pull data or save.

**Reply by airtower-luna** [reply, marked as answer]
That has nothing to do with the URL, and has been in force for a while now. You cannot use your password for `git push` or other Git operations (but you still need it to log in). You need to [create a personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) and enter that instead when Git asks for your password. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

**Reply by RVanVlietII** [reply]
Thank you so much!!!! It worked finally.

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:
